Shared (App)/Models/MethodInfo.plist

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<array>
+	<dict>
+		<key>Description</key>
+		<dict>
+			<key>StringFormat</key>
+			<string>Registered %@</string>
```

Comment:
What do you guys think of this approach? Maintaining a list of string formats per contract/methodId so we can display the tx input in a descriptive way, instead of dumping the method name and its arguments? But it's something we'd have to maintain over time...

Reply:
I think maintaining a list is a good idea. While it involves more upfront work, and the ongoing maintenance cost, it's the best approach we have so far to provide really clear, user-friendly descriptions of transactions.

One question: why a `plist`? Would it not be better to put this in Swift directly, perhaps a map from `address+method` to a callback? This way, we can not only return text, but also other pieces of UI (logos, links, perhaps even in-wallet actions?)

Reply:
In addition to our discussions, this approach is still a bit limited as it  assumes that the parameter value’s string representation can be readily displayed. However, in some cases we will need to do some further formatting, e.g. when the input parameter is a timestamp, we would like to format it in a readable way. One way we could handle this is to add an additional type argument which tells the client how to format the value. wdyt?

Reply:
Yea, I agree. What about using `String(format:_:)` for example? https://riptutorial.com/swift/example/6342/formatting-strings

Reply:
Right! So in the current PR `String(format:arguments:)` is used (see `TransactionInputParser`). So we should be able to handle different format specifiers. But if we wanted to do date formatting for example then string formats wouldn't be sufficient anymore. 

There are also a couple of other things to consider: 
- The format specifier used in the string format has to match the type of the decoded input param.
- The list of arguments provided in the plist/json to be in the right order. Also each argument in the plist/json needs to match the param name of the decoded input.

It's easy for errors to creep in here. When the list contains errors in the string format or the supplied argument names then it will be difficult to catch. We will need to think about how we can add further validation perhaps, and/or how to properly test updates in the string format list.